Section 5 comments

lachellel opened the pull request:
95 Comments reviewed
78 comments addressed with changes or rejected

Outstanding items:

- A number of comments repeat the need to address OCSP responses / responders separate from "the CA" requirements. In this scenario, we need to address differences in understanding and terminology from traditional federal PKI. For this CP, the CA is responsible for a requirement including the certificate status information regardless of how the underlying system is designed.
- Four comments regarding fire inspection and fire suppression
- Reconciling 5.4.1 requirements with 5.5.1 requirements

closes #397 closes #407 partial #468 closes #485 closes #237 closes #382 closes #381 closes #238 closes #384 closes #387 closes #388 closes #239 closes #389 closes #390 closes #391 closes #240 closes #292 closes #393 closes #257 closes #241 closes #396 closes #258 closes #293 closes #294 closes #295 closes #399 closes #400 closes #401 closes #403 closes #404 closes #243 closes #405 closes #406 closes #227 closes #375 closes #255 closes #310 closes #376 closes #377 closes #378 closes #232 closes #233 closes #234 closes #235 closes #485 closes #236 closes #237 closes #256 closes #381 closes #382

lachellel committed:
section 5.1 updates closes #227 partial #468 closes #375 minor update to definitions for qualified auditor

lachellel committed:
rewrote section 5.2.1 closes #255 closes #310 closes #376 closes #377 closes #378 closes #232 closes #233 closes #234 closes #235

lachellel committed:
updated trusted roles, addressed sections 5.3 through section 5.8 comments, closes #397 closes #407 partial #468 closes #485 closes #237 closes #382 closes #381 closes #238 closes #384 closes #387 closes #388 closes #239 closes #389 closes #390 closes #391 closes #240 closes #292 closes #393 closes #257 closes #241 closes #396 closes #258 closes #293 closes #294 closes #295 closes #399 closes #400 closes #401 closes #403 closes #404 closes #243 closes #405 closes #406

lachellel committed:
add clock changes

aligned with 6.8 
Updated to address comment in #469 "Section 5.4.1 does not identify clock change as auditable event."

lachellel committed:
updating duties of officer and audit administrator

for #454

lachellel commented:
@konklone if you have time for a quick review to merge?

Section 5, Section 6, Section 8 in the queue.  Working on Section 7 and incorporating the cert profiles.  Next up is a top to bottom for outstanding issues and TBDs.

konklone commented on certificate-policy.md:
```diff
@@ -841,289 +811,247 @@ All persons filling trusted roles shall be selected on the basis of loyalty, tru
+
+The CA shall undergo or perform a Vulnerability Scan:
+- At least once per quarter, on public and private IP addresses identified as within the CA's system boundaries
+- Within one week of receiving a request from the FPKIPA or the U.S. Government Federal Information Security Modernization Act Authorizing Official for the CA system or the CA/B Forum
```
I don't think the CA/B Forum is the right entity here. They don't play an active enforcement or oversight role on individual CAs, only root programs do. So I think we want to have this reference root program representatives, rather than the CA/B Forum.

lachellel replied:
I too questioned this one.  The original verbiage referencing CA/B Forum is in the Baseline Requirements; I updated for USG and authorities (FISMA and PA by delegation from CIO council).  I did not see any trust store program (from microsoft, mozilla, java, etc) policy that overrode the original Baseline Requirements.    

I'm marking this for review by general counsel.  We should also review the CA/B Forum bylaws to determine how a CA could be directed (e.g. through a Vote?) to perform penetration testing, and the legal jurisdiction. 

[I'm removing the CA/B Forum for now.]

konklone commented on certificate-policy.md:
```diff
@@ -841,289 +811,247 @@ All persons filling trusted roles shall be selected on the basis of loyalty, tru
 
 ### 5.5.2 Retention period for archive
-The CA shall retain all documentation relating to certificate requests and the verification thereof, and all Certificates and revocation thereof, for a minimum of 10 years and 6 months without any loss of data after any Certificate based on that documentation ceases to be valid.
+The CA shall retain all documentation relating to certificate requests and the verification thereof, and all Certificates and revocation information thereof, for a minimum of seven years without any loss of data after any Certificate based on that documentation ceases to be valid.
```
Is 7 years meant to reference current NARA rules? We may have an interest in keeping some data beyond 7 years, and referencing records schedules timelines implicitly emphasizes that deletion is default-encouraged/okay after that timeline. I think we would benefit from having a sentence here that says that after 7 years, any data deletion should still be approved by FPKIPA.

lachellel replied:
The update was to address #257 and inconsistencies identified by Scott Perry.  

-The 7 years is Baseline Requirements.  
-The original 10 years, 6 months was Federal Common Policy (or Federal Bridge Certificate Policy).  
-The 10 years, 6 months was rooted (pun intended) in 800-63 + Digital / Electronic Signatures + CFRs.  
- One archival reference: https://www.archives.gov/records-mgmt/policy/pki.html#4-3

The 10 years, 6 months wasn't applicable to the purpose of this infrastructure - so I reverted to the baseline requirements.  

I'm hesitant to include that any data deletion after 7 years requires approval by the PA, as a voting body, or directly by the PA co-chairs.  I think it is better to state the basic requirements, and allow any new NARA requirements or FISMA requirements to take precedence (and not require a policy to be changed or a vote / approval to be taken).  

Thoughts? We can open an issue to discuss but I want to merge!

konklone commented on certificate-policy.md:
I would rephrase to allow for redaction to be optional, and for the FPKI to publish additional copies on non-government websites (such as Bugzilla) or on non-web areas (such as an email to m.d.s.p). 

So for example:

> - Publicly publish a final incident report in one or more internet-accessible locations, with information redacted as necessary.

konklone reviewed (changes requested):
I have one substantive issue that I think needs change, and two substantive suggestions for other changes.

lachellel committed:
required penetration testing

and who can direct an out of cycle pen test to be performed; removing ca/b forum for now pending review by counsels

lachellel committed:
updated publishing of incident reports

per eric comment
